ddl/schema.go

```diff
@@ -109,6 +111,35 @@ func checkSchemaNotExistsFromStore(t *meta.Meta, schemaID int64, dbInfo *model.D
 	return nil
 }
 
+func onModifySchemaCharsetAndCollate(t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	var toCharset, toCollate string
+	if err := job.DecodeArgs(&toCharset, &toCollate); err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+
+	dbInfo, err := checkSchemaExistAndCancelNotExistJob(t, job)
+	if err != nil {
+		if infoschema.ErrDatabaseDropExists.Equal(err) {
+			desc := fmt.Sprintf("(Schema ID %d)", job.ID)
+			return ver, infoschema.ErrDatabaseNotExists.GenWithStackByArgs(desc)
+		}
+		return ver, errors.Trace(err)
+	}
+
```

Comment:
I think we can check "if dbInfo.Charset == toCharset && dbInfo.Charset == toCollate {return nil}" here。

Reply:
OK, then